meadow-client/src/Marlowe/Gen.purs

```diff
@@ -0,0 +1,124 @@
+module Marlowe.Gen where
+
+import Prelude
+
+import Control.Lazy (class Lazy, defer)
+import Control.Monad.Gen (class MonadGen, chooseInt)
+import Control.Monad.Rec.Class (class MonadRec)
+import Data.BigInteger (BigInteger)
+import Data.Newtype (wrap)
+import Data.NonEmpty ((:|))
+import Semantics (Contract(..), IdChoice, Observation(..), Value(..))
+
+import Control.Monad.Gen as Gen
+import Data.BigInteger as BigInteger
+
+genBigInteger :: forall m. MonadGen m => MonadRec m => m BigInteger
+genBigInteger = BigInteger.fromInt <$> chooseInt (negate 2147483648) 2147483647
+
+genIdChoice :: forall m. MonadGen m => MonadRec m => m IdChoice
+genIdChoice = do
+  choice <- genBigInteger
+  person <- genBigInteger
+  pure $ wrap { choice, person }
+
+lazyGenValue :: forall m. MonadGen m => MonadRec m => Lazy (m Value) => m Value
+lazyGenValue = defer \_ ->
+  genValue
+
+genValue :: forall m. MonadGen m => MonadRec m => Lazy (m Value) => m Value
+genValue = Gen.resize (min 5) $ Gen.sized genValue'
+  where
+  genValue' ::
+    Int ->
+    m Value
+  genValue' size
+    | size > 1 = Gen.resize (_ - 1) $ Gen.oneOf $ genLeaf :| [ NegValue <$> lazyGenValue
+                                                             , AddValue <$> lazyGenValue <*> lazyGenValue
+                                                             , SubValue <$> lazyGenValue <*> lazyGenValue
+                                                             , MulValue <$> lazyGenValue <*> lazyGenValue
```

Comment:
This is a sign that those people from the Church of Recursion Schemes will be visiting again soon. :-)